src/Data/List/NonEmpty/Base.agda

```diff
@@ -204,8 +204,7 @@ snocView (x ∷ .(xs List.∷ʳ y)) | xs List.∷ʳ′ y = (x ∷ xs) ∷ʳ′ y
 -- The last element in the list.
 
 last : List⁺ A → A
-last xs with snocView xs
-last .(ys ∷ʳ y) | ys ∷ʳ′ y = y
+last xs = case snocView xs of λ { (_ ∷ʳ′ y) → y}
```

Comment:
What are the trade-offs between irrefutable `with` and using `case_of_` with a(n irrefutable) pattern-matching lambda?

Reply:
Not entirely sure. What I'd really want here is a projector for `_∷ʳ′_`. So I've done that, which makes it all much clearer. (Makes me wish we could `derive` projectors for single-constructor `data` automatically!)